test/test-source-map-consumer.js

```diff
@@ -1119,3 +1119,16 @@ exports['test consuming names and sources that are numbers'] = function (assert)
   });
   assert.equal(i, 1);
 };
+
+exports['test from issue 227'] = function (assert) {
+  var consumer = new SourceMapConsumer({
+    version: 3,
+    sources: [ 'index.js' ],
+    names: [],
+    mappings: ';;AAAA,IAAI,OAAO,MAAP',
+    file: 'index.js',
+    sourceRoot: './src/',
+    sourcesContent: [ 'var name = "Mark"\n' ]
+  });
+  consumer.sourceContentFor(consumer.sources[0]);
```

Comment:
I'd also like to see an assertion for the new behavior you've introduced here, since the output will not be exactly the same.

```
assert.equal(consumer.sourceRoot, 'src/');
```